### PR TITLE
boost-json: Use latest builder

### PR DESCRIPTION
**Pull request by tyler92**
Latest builder and small refactoring for build.sh

### CODE REVIEW

**Comment by vitorguidi** [comment]
> tyler92 is a new contributor to [projects/boost-json](https://github.com//google/oss-fuzz/tree/master/projects/boost-json). The PR must be approved by known contributors before it can be merged. The past contributors are: maflcko, grisumbras, DonggeLiu, pauldreik

lgtm

@maflcko @grisumbras @pauldreik can I go ahead and merge this?

File: projects/boost-json/build.sh [hunk 1]
```diff
@@ -15,14 +15,21 @@
...
+     --prefix=$WORK/stage                                       \
+     --with-json                                                \
+     include=/usr/local/include/x86_64-unknown-linux-gnu/c++/v1 \
+     link=static                                                \
+     install
 
 for i in libs/json/fuzzing/*.cpp; do
    fuzzer=$(basename $i .cpp)
    $CXX $CXXFLAGS -pthread libs/json/fuzzing/$fuzzer.cpp -I $WORK/stage/include/ $WORK/stage/lib/*.a $LIB_FUZZING_ENGINE -o $OUT/$fuzzer
+   zip -q -r -j $OUT/${fuzzer}_seed_corpus.zip libs/json/fuzzing/old_crashes
```
Comment (grisumbras) [inline projects/boost-json/build.sh:33]: What is the purpose of this line?

Comment (tyler92) [reply projects/boost-json/build.sh:33]: Here we create a zip archive with a seed corpus for each fuzzer. This helps fuzzers generate valid inputs and reach more branches. Any samples, even old crashes, can be used for this purpose. In other words, this is for fuzzers' performance.

Comment (tyler92) [reply projects/boost-json/build.sh:33]: More info here: https://google.github.io/oss-fuzz/getting-started/new-project-guide/#seed-corpus

**Review by grisumbras — APPROVED** [review]
LGTM

**Review by grisumbras — APPROVED** [review]
LGTM

**Review by grisumbras — APPROVED** [review]
LGTM

File: projects/boost-json/build.sh (hunk 1 above)
Comment (grisumbras) [reply projects/boost-json/build.sh:33]: Got it.

**Review by vitorguidi — APPROVED** [review]
lgtm

**Pull request merged**